### PR TITLE
[2/N] Change static functions in headers to inline

### PR DESCRIPTION
**Pull request by cyyever**
Follows #127727

### CODE REVIEW

File: torch/csrc/utils/cuda_enabled.h [hunk 1]
```diff
@@ -1,15 +1,13 @@
 #pragma once
 
-namespace torch {
-namespace utils {
+namespace torch::utils {
 
-static inline bool cuda_enabled() {
+inline constexpr bool cuda_enabled() {
```
Comment (Skylion007) [inline torch/csrc/utils/cuda_enabled.h:5]: Not sure this should be constexpr as we may change it in the future to require some dynamic dispatch etc to determine if CUDA is really enabled.

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/utils/cuda_enabled.h (hunk 1 above)
Comment (cyyever) [reply torch/csrc/utils/cuda_enabled.h:5]: constexpr can contain non-constexpr expressions.

Comment (Skylion007) [reply torch/csrc/utils/cuda_enabled.h:5]: Yeah, but downstream libraries could break if they depend on it being constexpr and we revert it's constexpr-ness later.

**Pull request closed without merging**